src/business-logic/services/test/throttled-rundown-service.spec.ts

```diff
@@ -0,0 +1,124 @@
+import { ThrottledRundownService } from '../throttled-rundown-service'
+import { instance, mock, verify, when } from '@typestrong/ts-mockito'
+import { Rundown, RundownInterface } from '../../../model/entities/rundown'
+import { ThrottledRundownException } from '../../../model/exceptions/throttled-rundown-exception'
+
+describe(ThrottledRundownService.name, () => {
+  describe('when two operations are performed within the locked interval', () => {
+    beforeEach(() => jest.useFakeTimers())
+    afterEach(() => jest.useRealTimers())
+    describe('throws an error to second operation', () => {
+      it('throws error when two take nexts are performed with 0ms delay', async () => {
+        const mockRepo: ThrottledRundownService = mock<ThrottledRundownService>()
+        const randomRundownId: string = 'randomRundownId'
+        const randomRundown: Rundown = new Rundown({ id: randomRundownId } as RundownInterface)
+
+        when(mockRepo.takeNext(randomRundownId)).thenReturn(Promise.resolve())
+
+        const testee: ThrottledRundownService = new ThrottledRundownService(instance(mockRepo))
+
+        await testee.takeNext(randomRundown.id)
+
+        verify(mockRepo.takeNext(randomRundownId)).once()
+
+        const result: () => Promise<void> = () => testee.takeNext(randomRundown.id)
+
+        expect(result).toThrow(ThrottledRundownException)
+      })
+
+      it('throws error when two different actions are performed with 0ms delay', async () => {
+        const mockRepo: ThrottledRundownService = mock<ThrottledRundownService>()
```

Comment:
What does `mockRepo` mean? `repo` is useally short for `repository` which is what connects and talks to databases, but you are mocking `ThrottledRundownService` which is not a repository.

Also, the file you want to test is `ThrottledRundownService`. You should not be mocking the file under test.

I see why you have this mock (so you can give it to your testee), but your testee expects a `RundownService` so you can just mock the `RundownService` instead of the `ThrottledRundownSerivece`:
```
const mockRundownService: RundownService = mock<RundownService>()
```

This comment goes for all your tests.

Reply:
Fixed and add some more test cover logic out side of throttled mechanism